crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2414,6 +2462,8 @@ mod tests {
     #[test_case(Ty::Intersection{pos: vec![], neg: vec![Ty::IntLiteral(2)]}, Ty::Intersection{pos: vec![], neg: vec![Ty::BuiltinInstance("int")]})]
     #[test_case(Ty::BuiltinInstance("int"), Ty::Intersection{pos: vec![], neg: vec![Ty::IntLiteral(3)]})]
     #[test_case(Ty::IntLiteral(1), Ty::Intersection{pos: vec![Ty::BuiltinInstance("int")], neg: vec![Ty::IntLiteral(1)]})]
+    #[test_case(Ty::BuiltinClassLiteral("int"), Ty::BuiltinClassLiteral("str"))]
+    #[test_case(Ty::BuiltinInstance("int"), Ty::BuiltinClassLiteral("int"))]
```

Comment:
Another test I think might be worth having (since it's easy to get confused over) is that `BuiltinClassLiteral("int")` is not a subtype of `BuiltinClassLiteral("object")`. Though admittedly it's orthogonal to this PR, I just thought of it since you are adding subtype tests around class literal types :)